### DISCUSSION TITLE
Is it possible to await for interests on *multiple* AsyncFd?

### DISCUSSION THREAD

**Discussion by lestrozi** [tokio-rs/tokio#6296] [Q&A] [answered]
Hello  
I have multiple sockets open and I want to read UDP packets from any of them.  
A simple way to do that is to create one task per socket and `socket[i].readable().await` on each of them, but I'm wondering if it is possible to create a *single task* that register Interest::READABLE on all my sockets and gets awaken when *any* become readable?

Thanks!

**Comment by Darksonn** [comment]
You can do that with `tokio::join!` and other similar utilities in the futures crate, but spawning separate tasks will often perform better.

**Reply by lestrozi** [reply]
AFAIK `join!` will only return when **all** tasks are done, not when **any**. Did I get this wrong?

**Reply by Darksonn** [reply, marked as answer]
Right, I meant select.
